docs/runbooks.md

```diff
@@ -0,0 +1,147 @@
+# Rollout-operator runbooks
+
+## Helm chart
+
+A Helm chart is available [here](https://github.com/grafana/helm-charts/tree/main/charts/rollout-operator) for the `rollout-operator`.
+
+## Alerts
+
+### IncorrectWebhookConfigurationFailurePolicy
+
+This alert fires when one of the validating webhooks used by the `rollout-operator` has its failure policy set to `Ignore`.
+
+In normal operations the failure policy should be set to `Fail`.
+
+```
+kind: ValidatingWebhookConfiguration
+...
+  failurePolicy: Fail
+```
+
+How it **works**:
+
+- This alert checks on the configured failure policy of the `pod-eviction` and `zpdb-validation` validating webhooks via the `kube_validating_webhook_failure_policy` metric
+- Although it may be valid to temporarily enable an `Ignore` failure mode, normal operations should have the failure mode set to `Fail`
+- When in `Ignore` mode the Kubernetes API server ignores a webhook failure if the webhook can not be reached
+- This would result in the zone aware pod disruption budget not being enforced or the ZPDB configuration validator not being enforced if the `rollout-operator` is not running
+
+How to **investigate**:
+
+- Review the configuration of the `pod-eviction` and `zpdb-configuration` webhooks:
+  - `kubectl -n <namespace> get ValidatingWebhookConfigurations zpdb-validation-<namespace> -o yaml`
+  - `kubectl -n <namespace> get ValidatingWebhookConfigurations pod-eviction-<namespace> -o yaml`
+- Update the configuration to use a `Fail` policy. See jsonnet configuration options `ignore_rollout_operator_zpdb_eviction_webhook_failures` and `ignore_rollout_operator_zpdb_validation_webhook_failures` in [rollout-operator.libsonnet](https://github.com/grafana/rollout-operator/blob/main/operations/rollout-operator/rollout-operator.libsonnet).
+
+### BadZoneAwarePodDisruptionBudgetConfiguration
+
+This alert fires when the zone aware pod disruption budget configuration validating webhook observes an invalid ZPDB object.
+
+This indicates that a malformed configuration has been applied, or an invalid configuration already exists when the `rollout-operator` starts.
+
+How it **works**:
+
+- If the `zpdb-validation` `ValidatingWebhookConfiguration` has a failure policy set to `Fail`, then the invalid configuration will be rejected by this webhook. This alert will NOT be triggered in this circumstance
+- If the failure policy is set to `Ignore`, then an invalid configuration may have been applied. This can occur if the `rollout-operator` pod is not running and a `ZPDB` configuration is applied. With a failure policy set to `Ignore` the configuration will be accepted if the webhook is unavailable
+- If the `zpdb-validation` `ValidatingWebhookConfiguration` is not applied then an invalid configuration may be applied. This alert will trigger in this scenario.
```

Comment:
```suggestion
How it **works**:

- Under normal circumstances, there is a `zpdb-validation` validating webhook that should prevent ZPDBs with invalid configuration from being accepted by the Kubernetes control plane
- However, if the validating webhook did not reject the invalid ZPDB for some reason, then the rollout-operator will reject and ignore the invalid ZPDB, so the ZPDB will not be effective, and this alert will fire
- The invalid ZPDB could have been accepted for a number of reasons, including:
  - The validating webhook was not installed correctly when the invalid ZPDB was stored
  - The validating webhook's failure policy was set to `Ignore` instead of `Fail`
```